ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_dev/ExplicitDisclosureIT.scala

```diff
@@ -0,0 +1,417 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.suites.v1_dev
+
+import com.daml.error.definitions.LedgerApiErrors
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.Assertions._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.testtool.infrastructure.TransactionHelpers.createdEvents
+import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.daml.ledger.api.v1.command_service.SubmitAndWaitRequest
+import com.daml.ledger.api.v1.commands.{Command, DisclosedContract, ExerciseByKeyCommand}
+import com.daml.ledger.api.v1.event.CreatedEvent
+import com.daml.ledger.api.v1.transaction_service.GetTransactionsRequest
+import com.daml.ledger.api.v1.value.{Record, RecordField, Value}
+import com.daml.ledger.client.binding
+import com.daml.ledger.client.binding.Primitive
+import com.daml.ledger.test.model.Test._
+import com.daml.lf.value.Value.ContractId
+import scalaz.syntax.tag._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+// TODO ED: Enable in Sandbox-on-X conformance test-suite
+final class ExplicitDisclosureIT extends LedgerTestSuite {
+  import ExplicitDisclosureIT._
+
+  test(
+    "EDCorrectDisclosure",
+    "Submission works if the correct disclosure is provided",
```

Comment:
While test description says "submission works", i cannot understand what "works" mean here, especially with assertion of the error in the end.

Reply:
Lines 45 and 55 have successful submissions. The asserted errors pertains to a submission where the submitter doesn't have visibility. Would you improve it somehow?

Reply:
I like you renamed "works" to "successful" and it is more explicit :) As failure case is also part of this description I would add it there, but it makes description too long, so it's fine as is.